src/pineko/theory.py

```diff
@@ -469,42 +508,67 @@ def fk(self, name, grid_path, tcard, pdf):
                 f"= {fk_filename}\n",
                 f"with max_as={max_as}, max_al={max_al}, xir={xir}, xif={xif}",
             )
-            _grid, _fk, comparison = evolve.evolve_grid(
-                grid,
-                operators,
-                fk_filename,
-                max_as,
-                max_al,
-                xir=xir,
-                xif=xif,
-                assumptions=assumptions,
-                comparison_pdf=pdf,
-                meta_data={"theory_card": json.dumps(tcard)},
-            )
+
+            if n_ekos == 1:
+                _grid, _fk, comparison = evolve.evolve_grid(
+                    grid,
+                    operators1,
+                    fk_filename,
+                    max_as,
+                    max_al,
+                    xir=xir,
+                    xif=xif,
+                    assumptions=assumptions,
+                    comparison_pdf1=pdf1,
+                    meta_data={"theory_card": json.dumps(tcard)},
+                )
+            else:
+                with eko.EKO.edit(eko_tmp_path_b) as operators2:
+                    _grid, _fk, comparison = evolve.evolve_grid(
+                        grid,
+                        operators1,
+                        fk_filename,
+                        max_as,
+                        max_al,
+                        xir=xir,
+                        xif=xif,
+                        assumptions=assumptions,
+                        operators2=operators2,
+                        comparison_pdf1=pdf1,
+                        comparison_pdf2=pdf2,
```

Comment:
I'm wondering if we shouldn't add a warning at this level whenever the two PDFs are different. Ideally, the warning should print out the order of the convolution types which would provide some hints to the user on the order of the PDFs.

Reply:
I think we are doing it here: 

https://github.com/NNPDF/pineko/blob/4859448f78230f02f9dba11c0e22c07ca46280b2/src/pineko/comparator.py#L59-L65